### DISCUSSION TITLE
Documentation for "open" and "save"?

### DISCUSSION THREAD

**Discussion by carrete** [SVG-Edit/svgedit#455] [General]
I'm incorporating svgedit into a react app, so we'll be using an iframe to embed svgedit, and postMessage to communicate with svgedit. Unfortunately, I've so far been unable to find the documentation I need to accomplish this. The jsdoc links in the README of this repo return 404. I've looked through the source code and I see a (very nice) API that wraps the postMessage details away via callbacks, but I haven't yet found what I need to pass to this API. I assume I need to pass the drawing to be opened as a (possibly encoded) string of some sort sent via an "open" message, and expect the updated drawing to be returned as a (possibly encoded) string in the call back of a "save" message. Provided this is correct, what exactly would this look like in practice? Thanks in advance for any help. Sorry if I missed these answers in a different set of documentation.

**Comment by carrete** [comment]
I built the docs locally and I think I found what I need, namely:

```js
// Initialize the magic with:
const svgCanvas = new EmbeddedSVGEdit(window.frames.svgedit);

// Pass functions in this format:
svgCanvas.setSvgString('string');

// Or if a callback is needed:
svgCanvas.setSvgString('string')(function (data, error) {
     if (error) {
     // There was an error
     } else {
     // Handle data
     }
});
```

The string is not encoded, and `getSvgString` is used to read back the svg drawing.

**Reply by jfhenon** [reply]
OK good actually. For information, we also want to use svgedit in a react project. You may, by the way, look at the https://github.com/SVG-Edit/svg-edit-react to find a sample code using directly the canvas layer. 
We are in the process to update svgedit mainly to stop using old jQuery versions (jQuery is not so React friendly by the way). All contributions are welcome.

**Reply by carrete** [reply]
Thank @jfhenon . We saw that, and that's definitely what we would prefer to use. But I only see three commits made this past August. To me that says it's not quite ready for prime-time, or no?

**Reply by jfhenon** [reply]
No, it's not ready. Actually, it's really only a proof of concept now with no plan to go further at this time. We are in the process to refresh and use the original project like you.
